Mail footer notified partners

JonathanNEMRY opened the pull request:
This module adds into the footer the partners' name notified by this email

Commit:
[NEW][mail_footer_notified_partners]
A new module to include into the footer the partners' name notified by email

Commit:
[mail_footer_notified_partners][IMP] code review

Cedric-Pigeon commented:
@gurneyalex I confirm that @JonathanNEMRY is my colleague and part of Acsone and therefore have an Enterprise CLA signed.

gurneyalex commented:
@Cedric-Pigeon OCA database updated. Thanks!

Cedric-Pigeon commented:
@gurneyalex  Thanks

Cedric-Pigeon commented:
:+1:  Code and tests

gurneyalex commented on mail_footer_notified_partners/models/mail_followers.py:
```diff
@@ -0,0 +1,75 @@
+from openerp import tools
+from openerp.tools.translate import _
+
+ADDITIONAL_FOOTER = _('This message was also sent to')
```
Could you move the colon from line 55 inside the translation string? 

this will make things easier e.g. for the french translation where colons are preceeded by a space.

gurneyalex commented on mail_footer_notified_partners/models/mail_followers.py:
```diff
@@ -0,0 +1,75 @@
+            ]
+            additional_footer =\
+                '<br /><small>%s: %s</small><br />' %\
+                (ADDITIONAL_FOOTER, ', '.join(partners_name))
```
please use unicode strings. 

Also should there not be some kind of HTML sanitizing of the partner names / emails?

gurneyalex commented on mail_footer_notified_partners/__openerp__.py:
```diff
@@ -0,0 +1,39 @@
+    'summary': """
+        This module adds into the footer the partner's name notified by this
+        email""",
+    'author': "ACSONE SA/NV",
```
could you please add "Odoo Community Association (OCA)" to the author entry (by appending it to the string, separated by a comma)? This allows some visibility of OCA addons on apps.odoo.com.

gurneyalex commented:
@JonathanNEMRY thanks for this useful addon. I've made a review, and there are a couple of things to update, but nothing serious. :smile_cat:

Commit:
[mail_footer_notified_partners]
[CHG]
* Add "Odoo Community Association (OCA)" in manifest's author
* Use unicode
* Move ": " into the translatable string

JonathanNEMRY commented:
@gurneyalex 
Thanks for the review!! I think it is OK now.
About the HTML sanitizing I'm not sure it is required (I've just made this like Odoo into their footer signature). But feel free to suggest anything you think better :)

JonathanNEMRY commented:
@gurneyalex: Is the label `needs fixing` still required?

hbrunn commented on mail_footer_notified_partners/models/mail_followers.py:
```diff
@@ -0,0 +1,75 @@
+from openerp import tools
+from openerp.tools.translate import _
+
+ADDITIONAL_FOOTER = _('This message was also sent to: ')
```
this will give you the wrong translation because the translation is not executed in the user's context but evaluated on load. Better put this into the code

Commit:
[FIX] Include the string into the source code to have a correct transaltion

hbrunn commented:
:+1:

sbidoul commented on mail_footer_notified_partners/README.rst:
```diff
@@ -0,0 +1,35 @@
+Mail Footer Notified Partners
+=============================
+
+This module adds into the footer the partner's name notified by this email
```
This module adds the list of notified partners in the footer of notification e-mails sent by Odoo.

Commit:
[IMP] Use a better description of the module

sbidoul commented:
@JonathanNEMRY while testing I got this message

```
Sent by ACSONE SA/NV using Odoo 
This message was also sent to: Bidoul, Stéphane
about Expense XYZ
```

while this would have been nicer:

```
Sent by ACSONE SA/NV using Odoo about Expense XYZ
This message was also sent to: Bidoul, Stéphane
```

is it feasible?

JonathanNEMRY commented:
@sbidoul it is due to the `_get_partner_access_link`  from `Portal`  (https://github.com/odoo/odoo/blob/8.0/addons/portal/mail_mail.py)
To set `about Model Record` before the custom footer it should have a second module that depends on `mail_footer_partner_notifed` and `portal` in order to control the inheritance chain. And the current module should be refactored to use the same `_get_partner_access_link` method (`instead of get_signature_footer`)

jgrandguillaume commented:
:+1:

JonathanNEMRY commented:
@sbidoul An idea should be to add the notifed partners at the beginning of the footer.
I think the problem could be easily solve with this workaround. If you agree with this solution I can implement it before the merge of this PR

sbidoul commented:
@JonathanNEMRY yes I still believe the text as shown in https://github.com/OCA/server-tools/pull/147#issuecomment-95490738 is not very nice to read. So putting the notified partner before "Sent by Odoo" could be a good solution.

Commit:
[mail_footer_notifed_partners]
[CHG]
Add the additional footer before the footer to avoid conflict of layout with portal and/or others inherited behavior

sbidoul commented:
With the last commit 81776f7, the notified partners come before the signature block.

IMHO, it's looking better.

:+1:

sbidoul committed:
mail_footer_notified_partner: normalize version number

sbidoul commented:
superseded by https://github.com/OCA/social/pull/37
